client/hooks/useMatchMedia.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+
+const useMatchMedia = (breakpoint: number) => {
+  const query = matchMedia(`(max-width: ${breakpoint}px)`);
+  const [matching, setMatching] = useState(query.matches);
```

Comment:
Since the matching state depends on accessing a window property, we might think about lazy loading the initial state to make sure everything runs once on initial render. This might be overkill given that this hook probably won't run that often anyway.

Reply:
@colinvandergraaf I'm not sure I understand the intent here. Is the concern that window size may not be available when the component renders? Can you point to any reference that says this is something the app should manage?